Kitodo-SRU-Import/src/main/java/org/kitodo/sruimport/SRUImport.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.sruimport;
+
+import static org.apache.http.HttpStatus.SC_OK;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.api.externaldatamanagement.ExternalDataImportInterface;
+import org.kitodo.api.externaldatamanagement.SearchResult;
+import org.kitodo.config.OPACConfig;
+import org.w3c.dom.Document;
+
+public class SRUImport implements ExternalDataImportInterface {
+
+    private static final Logger logger = LogManager.getLogger(SRUImport.class);
+    private static final String NAME_ATTRIBUTE = "[@name]";
+    private static final String VALUE_ATTRIBUTE = "[@value]";
+    private static final String LABEL_ATTRIBUTE = "[@label]";
+
+    private static String protocol;
+    private static String host;
+    private static String path;
+    private static LinkedHashMap<String, String> parameters = new LinkedHashMap<>();
+    private static HashMap<String, String> searchFieldMapping = new HashMap<>();
+    private static String equalsOperand = "==";
+    private static HttpClient sruClient = HttpClientBuilder.create().build();
+
+    /**
+     * Standard constructor.
+     */
+    public SRUImport() {
+        // TODO: implement SchemaConverter and instantiate here
+    }
+
+    @Override
+    public Document getFullRecordById(String catalogId, String id) {
+        loadOPACConfiguration(catalogId);
+        LinkedHashMap<String, String> queryParameters = new LinkedHashMap<>(parameters);
+        try {
+            URI queryURL = createQueryURI(queryParameters);
+            return performQueryToDocument(queryURL.toString() + "&query=ead.id==" + id);
+        } catch (URISyntaxException e) {
+            logger.error(e.getLocalizedMessage());
+        }
+        // TODO: transform hit to Kitodo internal format using SchemaConverter!
+        return null;
+    }
+
+    @Override
+    public SearchResult search(String catalogId, String field, String term, int rows) {
```

Comment:
Regarding my question above... you just put the search field in a map and call the other method ;)